Upgrade to Filament v3

dan-tang-ssd opened the pull request:
This PR is submitted to upgrade filament from v2 to v3.

This PR is working in progress, not yet ready for review.

I did below items so far:
1. Follow filament v3 upgrade guide
2. Remove conflict packages from composer.json
3. Install filament v3
4. Install previously removed conflict packages one by one
5. P.S. v2 grouped sidebar is not supported in v3

TODO:
1. Upgrade laravel from v9 to v10

dan-tang-ssd committed:
Upgrade filament from v2 to v3. TODO: Upgrade laravel from v9 to v10

dan-tang-ssd committed:
Upgrade Laravel from v9 to v10. Removed some packages during upgrade process

dan-tang-ssd commented:
Hi Dave,

I think this PR is now ready for your review, and also need your help...

---

I have performed below items:
1. Upgrade Laravel from v9 to v10
2. Some packages are removed during upgrade process, I re-installed them afterwards if necessary
3. Some packages are deprecated or replaced by Laravel framework, I changed to use the new class name
4. In filament v3, I added back the original groupings to filament resources

---

Questions:
1. I am not able to find Roles and Permissions CRUD panels, are they belong to backpack? And how to add them back?
2. The filament v3 resources page layout looks a bit strange, it is quite different when compared with Tape GIZ... Is there anything we can do to beautify it?

---

Screen shots:

Metrics Library
![image](https://github.com/stats4sd/tpp-metrics-library/assets/86968034/5088a09d-5975-44ec-b11b-7f535488b3bb)


Tape GIZ
![image](https://github.com/stats4sd/tpp-metrics-library/assets/86968034/ac0091db-11d2-44e4-bde7-cca2d453ecd8)

dave-mills commented:
Yes, I think the roles and permission pages were from the Backpack side of this system. (We tried out both before settling on Filament for the better relationship-managers). We can add them in pretty quickly once we've got the upgrade finished.

You're right about the UI - that's pretty odd. I assume there are some things conflicting with the 'default' Filament. Perhaps some blade files that we can remove, or some css. I'll have a look.

dave-mills committed:
remove call to old filament theming

dave-mills committed:
add filament:upgrade call to composer:
- This ensures filament's front-end assets and configs are republished after each composer install

dave-mills committed:
Test all resource panels:
- Removed custom form components that used v2 templates as they were causing issues
- Minor bug fixes: rename metric_users  to ; resolve breaking issues with custom discussion point creation due to upgrade;
- deleted old vendor views folder to ensure they aren't interfering

dave-mills reviewed (approved):
@dan-tang-ssd  - The UI issues were caused by the fact it was trying to call a v2 theme that was setup in the AppProvider. Removing that brought it back to the 'default' Filament 3 look. 

I also made a few other updates:

- I added the `php artisan filament:upgrade` call to composer.json, as that needs to be run after each composer update to ensure we have the latest assets published. 
- I tested all the panels - only quickly; just to make sure nothing breaks immediately when I try to view or edit an entry. I made a bunch of minor fixes as lots of places were referencing custom versions of the Filament form components, which were of course all using the V2 templates that are now broken. 

(All the custom components were entirely to add 'suffixActions' to specific components that don't have them by default, e.g. to allow adding the discussion point button to text areas or table repeaters. Not sure we need the disccussion points going forward, so I've removed the custom components instead of updating them). 


I think this is good to merge now - It looks like a functional Filament 3 app :). Cheers!
